tests/test_nodes_utils.py

```diff
@@ -40,3 +48,105 @@ def test_consume_common_prefix(left, right, expected):
     expected_b = (expected[0], expected[2], expected[1])
     assert actual_a == expected
     assert actual_b == expected_b
+
+
+@pytest.mark.parametrize(
+    'valid_node',
+    (
+        (b"\x00\x03\x04\x05\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x02value"),  # noqa: E501
+    ),
+)
+def test_binary_trie_node_parsing_with_valid_input(valid_node):
+    parse_node(valid_node)
```

Comment:
What do you think about adding assertions that the return of this function is returning the expected values?  Seems like a small addition that would improve this test quite a bit.